WooCommerce/src/androidTest/kotlin/com/woocommerce/android/e2e/screens/shared/FilterScreen.kt

```diff
@@ -40,11 +54,24 @@ class ProductFilterScreen : Screen {
         return ProductListScreen()
     }
 
-    fun leaveFilterScreen(): ProductListScreen {
+    fun tapShowOrders(): OrderListScreen {
+        clickOn(R.id.showOrdersButton)
+        return OrderListScreen()
+    }
+
+    fun leaveFilterScreenToProducts(): ProductListScreen {
         if (isElementDisplayed(R.id.filterList) || isElementDisplayed(R.id.filterOptionList)) {
             tapShowProducts()
         }
 
         return ProductListScreen()
     }
+
+    fun leaveFilterScreenToOrders(): OrderListScreen {
+        if (isElementDisplayed(R.id.filterList)) {
+            tapShowOrders()
+        }
+
+        return OrderListScreen()
+    }
```

Comment:
A suggestion on this, we could potentially combine this into a generic method, too (and expand it as needed, I know that filter is available in coupons as well, maybe payments next) to something like:

```suggestion
    fun leaveFilterScreen(toScreen: String): Screen {
        if (isElementDisplayed(R.id.filterList) || isElementDisplayed(R.id.filterOptionList)) {
            when (toScreen) {
                "Products" -> {
                    tapShowProducts()
                    return ProductListScreen()
                }
                "Orders" -> {
                    tapShowOrders()
                    return OrderListScreen()
                }
                else -> print("Screen not defined!")
            }
        }
        return this
    }
```

this way, we would be able to know from the function itself the screens that has filter available

Reply:
This is a good suggestion, @jostnes, that I did not think about initially!

However, I decided to leave it unchanged because:

1. The proposed function has multiple return values, that are based on hardcoded strings. While having multiple return values is not inherently bad, I think it complicates the process of understanding the code.
2. By having dedicated functions, we show in autocomplete the available options, the user doesn't need to know which string they should use, they can't make a mistake.
3. The resulting code does not make the code shorter or easier to visually parse.
4. The most impactful, IMO: the logic of the proposed function is different. If the original functions, the "parent" screen would be attempted for return no matter what (means, we return "Products List" no matter if we saw/tapped "Show products" or not) which made a meaningful use case, in case if we don't know if filter screen was open, but we know that we generally work with Products List in the test. In the proposed function, if we don't see the filter screen, we return the filter screen?

Reply:
Perhaps I could have written that example better. The idea of that is to have a generic function that can be used for a feature that is available on more than one screen (this way, we can ensure consistent naming and style when trying to do the same thing across multiple tests). When only two screens are using it, I agree that it does not shorten the code. Still, the main point is not to make it shorter but to group it together as they have a similar function (also, when more screens are using it, it will eventually be shorter). For point 2, that can be easily remedied by adding a comment to the function like the one below so that it shows up when you hover over it:
```
...
* @param toScreen Accepted values: Products, Orders
...
```

The else statement could probably have been better if it threw an error so that when the condition is not met, the test can fail fast (e.g. if test looking is looking for that on the Coupons screen, but if it's not supported yet, fail it). The proposed solution, the return value is Screen, so if that condition is not met, it will return the screen the test was on.

The current implementation works, too, so no changes are needed on that 👍